src/main/java/io/xstefank/wildfly/bot/ConfigFileChangeProcessor.java

```diff
@@ -46,14 +50,20 @@ void onFileChanged(@PullRequest.Opened @PullRequest.Edited @PullRequest.Synchron
                     Optional<WildFlyConfigFile> file = Optional.ofNullable(yamlObjectMapper.readValue(updatedFileContent, WildFlyConfigFile.class));
 
                     if (file.isPresent()) {
-                        githubCommitProcessor.commitStatusSuccess(pullRequest, CHECK_NAME, "Valid");
-                        Log.debug("Configuration File check successful");
+                        List<String> problems = validateFile(file.get());
+                        if (problems.isEmpty()) {
+                            githubCommitProcessor.commitStatusSuccess(pullRequest, CHECK_NAME, "Valid");
+                            Log.debug("Configuration File check successful");
+                        } else {
+                            githubCommitProcessor.commitStatusError(pullRequest, CHECK_NAME, String.join(",", problems));
+                            Log.warnf("Configuration File check unsuccessful. %s", String.join(",", problems));
```

Comment:
nothing big, but if you do the same operation in two lines after each other it's better to extract the result. I don't expect problems will be a large list but still.